scripts/estimation.ts

```diff
@@ -0,0 +1,208 @@
+  await estimateContractDeployment('Outbox', nodeInterface, [])
+
+  await estimateContractDeployment('ERC20Bridge', nodeInterface, [])
+  await estimateContractDeployment('ERC20Inbox', nodeInterface, [maxDataSize])
+  await estimateContractDeployment('ERC20RollupEventInbox', nodeInterface, [])
+  await estimateContractDeployment('ERC20Outbox', nodeInterface, [])
+
+  await estimateContractDeployment('BridgeCreator', nodeInterface, [
+    [
+      ethers.Wallet.createRandom().address,
+      ethers.Wallet.createRandom().address,
+      ethers.Wallet.createRandom().address,
+      ethers.Wallet.createRandom().address,
+      ethers.Wallet.createRandom().address,
+    ],
+    [
+      ethers.Wallet.createRandom().address,
+      ethers.Wallet.createRandom().address,
+      ethers.Wallet.createRandom().address,
+      ethers.Wallet.createRandom().address,
+      ethers.Wallet.createRandom().address,
+    ],
+  ])
+  await estimateContractDeployment('OneStepProver0', nodeInterface)
+  await estimateContractDeployment('OneStepProverMemory', nodeInterface)
+  await estimateContractDeployment('OneStepProverMath', nodeInterface)
+  await estimateContractDeployment('OneStepProverHostIo', nodeInterface)
+  await estimateContractDeployment('OneStepProofEntry', nodeInterface, [
+    ethers.Wallet.createRandom().address,
+    ethers.Wallet.createRandom().address,
+    ethers.Wallet.createRandom().address,
+    ethers.Wallet.createRandom().address,
+  ])
+  await estimateContractDeployment('ChallengeManager', nodeInterface)
+  await estimateContractDeployment('RollupAdminLogic', nodeInterface)
+  await estimateContractDeployment('RollupUserLogic', nodeInterface)
+  await estimateUpgradeExecutorDeployment(nodeInterface)
+  await estimateContractDeployment('ValidatorUtils', nodeInterface)
+  await estimateContractDeployment('ValidatorWalletCreator', nodeInterface)
+  await estimateContractDeployment('RollupCreator', nodeInterface)
+  await estimateContractDeployment('DeployHelper', nodeInterface)
+}
+
+function _handleGasEstimates(
+  gasEstimateComponents: [BigNumber, BigNumber, BigNumber, BigNumber] & {
+    gasEstimate: BigNumber
+    gasEstimateForL1: BigNumber
+    baseFee: BigNumber
+    l1BaseFeeEstimate: BigNumber
+  },
+  contractName: string
+) {
+  totalGas = totalGas.add(gasEstimateComponents.gasEstimate)
+  totalL1Gas = totalL1Gas.add(gasEstimateComponents.gasEstimateForL1)
+  totalL2Gas = totalL2Gas.add(
+    gasEstimateComponents.gasEstimate.sub(
+      gasEstimateComponents.gasEstimateForL1
+    )
+  )
+  l2BaseFee = gasEstimateComponents.baseFee
+  l1BaseFee = gasEstimateComponents.l1BaseFeeEstimate
+
+  const P = l2BaseFee
+  const L1P = l1BaseFee.mul(16)
+  const l1Size = gasEstimateComponents.gasEstimateForL1.mul(P).div(L1P)
+  const L1C = L1P.mul(l1Size)
+  const B = L1C.div(P)
+  const G = gasEstimateComponents.gasEstimate
+    .sub(gasEstimateComponents.gasEstimateForL1)
+    .add(B)
+  const TXFEES = P.mul(G)
```

Comment:
This simply reduce into `TXFEES = P * gasEstimateComponents.gasEstimate`, which seems to be a convoluted way of doing simple gas estimation. This should instead take an extra parameter to allow specifying target L1 gas price, and scale L1C based on that e.g. `L1C = L1P.mul(targetL1Fee).div(l1BaseFee).mul(l1Size)`